miner/worker.go

```diff
@@ -481,6 +484,43 @@ func (miner *Miner) fillTransactions(interrupt *atomic.Int32, env *environment)
 			return err
 		}
 	}
+
+	// Filter out inclusion list transactions that are not included in the block.
+	isIncludedTx := make(map[common.Hash]bool)
+	for _, tx := range env.txs {
+		isIncludedTx[tx.Hash()] = true
+	}
+
+	pendingInclusionListTxs := make([]*types.Transaction, 0)
+	for _, tx := range env.inclusionList {
+		if isIncludedTx[tx.Hash()] {
+			continue
+		}
+
+		pendingInclusionListTxs = append(pendingInclusionListTxs, tx)
+	}
+
+	// Append any valid inclusion list transactions at the end of the block.
```

Comment:
We should sort IL+pool transactions by how many fees they pay and use this to prioritize inclusion.  It's possible there could be high-paying txs in the IL that aren't in our pool.

Reply:
Absolutely. Current implementation is a naive one, prioritized easier implementation and testing. Do you think we need to adapt the inclusion strategy in this PR?

Reply:
It's not absolutely vital, but we will want to make the change before this goes into production.

Is there anything else that you omitted in this PR to keep the diff easier to read/test?

Reply:
Nothing comes to my mind at the moment. Whether to use `engine_updatePayloadWithInclusionListV1` or `engine_forkchoiceUpdatedV4` could be changed in future as it's not decided yet.

Reply:
I've come up with another one. It takes a naive approach when building an IL, too. It doesn't use any heuristic algorithm such as taking the time spent in the mempool into account, and so on.

Reply:
Like I wrote on the call:  because it's unlikely that we will simultaneously be an IL committee member and the proposer for the same block, maybe we can afford to have a more altruistic default strategy than just building the most profitable IL.

Reply:
and also to build the most profitable IL, we would have to do full transaction execution here.  and even then, it's still speculative because we are not building the IL on top of the state that it will be applied.